src/gordon_gcp/clients/http.py

```diff
@@ -142,42 +142,52 @@ def __init__(self, auth_client=None, session=None):
-                                f' returned requesting {resp.url}: '
-                                f'{resp.reason}')
-                if token_refresh_attempts:
-                    logging.info(f'[{request_id}] Attempting request to '
-                                 f'{resp.url} again.')
-                    return await self.request(
-                        method, url,
-                        token_refresh_attempts=token_refresh_attempts,
-                        request_id=request_id,
-                        **req_kwargs)
-
-                logging.warning(f'[{request_id}] Max attempts refreshing auth '
-                                f'token exhausted while requesting {resp.url}')
-
-            # avoid leaky abstractions and wrap http errors with our own
-            try:
-                resp.raise_for_status()
-            except aiohttp.ClientResponseError as e:
-                msg = f'[{request_id}] Issue connecting to {resp.url}: {e}'
-                logging.error(msg, exc_info=e)
-                raise exceptions.GCPHTTPResponseError(msg, resp.status)
-
-            return await resp.text()
+        try:
+            async with self._session.request(method, url, **req_kwargs) as resp:
+                log_kw = {
+                    'request_id': request_id,
+                    'method': method.upper(),
+                    'url': resp.url,
+                    'status': resp.status,
+                    'reason': resp.reason
+                }
+                logging.debug(_utils.RESP_LOG_FMT.format(**log_kw))
+
+                if resp.status in REFRESH_STATUS_CODES:
+                    logging.warning(
+                        f'[{request_id}] HTTP Status Code {resp.status}'
+                        f' returned requesting {resp.url}: {resp.reason}')
+                    if token_refresh_attempts:
+                        logging.info(
+                            f'[{request_id}] Attempting request to {resp.url} '
+                            'again.')
+                        return await self.request(
+                            method, url,
+                            token_refresh_attempts=token_refresh_attempts,
+                            request_id=request_id,
+                            **req_kwargs)
+
+                    logging.warning(
+                        f'[{request_id}] Max attempts refreshing auth token '
+                        f'exhausted while requesting {resp.url}')
+
+                # avoid leaky abstractions and wrap http errors with our own
+                try:
+                    resp.raise_for_status()
+                except aiohttp.ClientResponseError as e:
+                    msg = f'[{request_id}] Issue connecting to {resp.url}: {e}'
+                    logging.error(msg, exc_info=e)
+                    raise exceptions.GCPHTTPResponseError(msg, resp.status)
+
+                return await resp.text()
+        except Exception as e:
```

Comment:
What do you think of this:

```py
try:
    async with self._session.request(method, url, **req_kwargs) as resp:
        # logic
       resp.raise_for_status()  # no wrapping here
       return await resp.text()
except aiohttp.ClientResponseError as e:
    # move the raise_for_status error handling to here; 
    # should have access to the `resp` variable
except Exception as e:
    # all other exception handling (no access to `resp` variable)
```

In essence, moving the inner `try`/`except` to the outer `try`/`except`.

Reply:
I considered that, but is it separating functions too much?  The handler has stuff about status that might be easier to follow where it is now.

Reply:
I'm open to either way though.

Reply:
I see the code's changed a bit since the last time I saw it, but if @econchick 's comment is agains the code as it is now, I prefer her suggestion, since it reduces the nesting of the try blocks and having to check `isinstance`

Reply:
ok, done.